security/pkg/pki/util/generate_cert.go

```diff
@@ -300,7 +304,7 @@ func genCertTemplateFromCSR(csr *x509.CertificateRequest, subjectIDs []string, t
 	return &x509.Certificate{
 		SerialNumber:          serialNum,
 		Subject:               subject,
-		NotBefore:             now,
+		NotBefore:             now.Add(-ClockSkewGracePeriod),
 		NotAfter:              now.Add(ttl),
```

Comment:
The clock skew can be symmetrical so if NotBefore is added with ClockSkewGracePeriod, should ClockSkewGracePeriod be added to NotAfter too (i.e. NotAfter becomes now.Add(ttl) + ClockSkewGracePeriod)?

Reply:
We already have a much larger grace period on the NotAfter. For example, if you have an hour lifetime (extremely short already), we would renew it at 30mins.